### PR TITLE
8298173: GarbageCollectionNotificationContentTest test failed: no decrease in Eden usage

### PR DESCRIPTION
**Pull request by kevinjwalls**
Test should not check that Eden has decreased in used size if the GCCause is "No GC".

These happen with the message "end of concurrent GC pause" which is logged by the GCMemoryManager _conc_gc_memory_manager (in G1MonitoringSupport).  This GCMemoryManager only interacts with the old gen memory pool, so it is expected that eden usage does not change.

Updated a few variable names to make it more obvious what count and number are recording, and some formatting.  The key  part of the change is at line 137 onwards.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8298173](https://bugs.openjdk.org/browse/JDK-8298173): GarbageCollectionNotificationContentTest test failed: no decrease in Eden usage

### Reviewers
 * [Daniel D. Daugherty](https://openjdk.org/census#dcubed) (@dcubed-ojdk - **Reviewer**) ⚠️ Review applies to [71607383](https://git.openjdk.org/jdk/pull/11572/files/716073838edb0b6fcf130fed5de685328f7b2a33)
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**) ⚠️ Review applies to [71607383](https://git.openjdk.org/jdk/pull/11572/files/716073838edb0b6fcf130fed5de685328f7b2a33)
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**) ⚠️ Review applies to [71607383](https://git.openjdk.org/jdk/pull/11572/files/716073838edb0b6fcf130fed5de685328f7b2a33)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11572/head:pull/11572` \
`$ git checkout pull/11572`

Update a local copy of the PR: \
`$ git checkout pull/11572` \
`$ git pull https://git.openjdk.org/jdk pull/11572/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11572`

View PR using the GUI difftool: \
`$ git pr show -t 11572`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11572.diff">https://git.openjdk.org/jdk/pull/11572.diff</a>

</details>

### CODE REVIEW

**Review by dcubed-ojdk — APPROVED** [review]
The variable renaming and code clean ups were a bit distracting.
However, I'm okay with the changes to skip verification when
"No GC" is the GCCause.

What kind of testing was done?

The bug is targeted at '21'. If you plan to fix this in '20' before
the cutoff, then please change the target to '20'. If you don't
plan to fix this in '20', then you'll need to ProblemList the test
in '20' because this failure is rather noisy as we already have
6 sightings in the JDK20 CI.

**Review by plummercj — APPROVED** [review]

**Comment by dcubed-ojdk** [comment]
This bug appears to be a regression caused by the fix for:

[JDK-8297247](https://bugs.openjdk.org/browse/JDK-8297247) Add GarbageCollectorMXBean for Remark and Cleanup pause time in G1

so I've bumped the priority from P4 -> P2 since it appears to be a regression.

File: test/jdk/com/sun/management/GarbageCollectorMXBean/GarbageCollectionNotificationContentTest.java [hunk 1]
```diff
@@ -85,16 +85,14 @@ public static void main(String[] args) throws Exception {
...
+        final ObjectName gcMXBeanPattern = new ObjectName("java.lang:type=GarbageCollector,*");
+        Set<ObjectName> names = mbs.queryNames(gcMXBeanPattern, null);
         if (names.isEmpty())
             throw new Exception("Test incorrect: no GC MXBeans");
-        number = names.size();
+        numberOfGCMBeans = names.size();
         for (ObjectName n : names) {
-            if(mbs.isInstanceOf(n,"javax.management.NotificationEmitter")) {
-                listenerInvoked.put(n.getCanonicalName(),null);
+            if(mbs.isInstanceOf(n, "javax.management.NotificationEmitter")) {
```
Comment (sspitsyn) [inline test/jdk/com/sun/management/GarbageCollectorMXBean/GarbageCollectionNotificationContentTest.java:94]: Nit: missed space in "if(".

**Review by sspitsyn — APPROVED** [review]
Looks good.
Posted one nit.
Thanks,
Serguei

File: test/jdk/com/sun/management/GarbageCollectorMXBean/GarbageCollectionNotificationContentTest.java (hunk 1 above)
Comment (kevinjwalls) [reply test/jdk/com/sun/management/GarbageCollectorMXBean/GarbageCollectionNotificationContentTest.java]: got it 8-)

**Comment by kevinjwalls** [comment]
Thanks for the reviews, thanks @dcubed-ojdk  for the pointer to the other change - this did not start failing by coincidence, there are new notifications happening.

It is hard to reproduce the failure, and they are mostly on linux aarch64, but I've seen one locally on linux x64. 
 The failures show an event where Eden size does not change: the test says that eden before size <= after size is a failure.

I cannot reproduce the problem with this change in mach5 or locally, in many runs.

Yes, the cleanup changes are distracting, but "count"?  That should be more meaningful. 8-)

**Comment by dcubed-ojdk** [comment]
@kevinjwalls - Thanks for getting the fix integrated into JDK20.
The CI will be less noisy today!

**Pull request closed without merging**